woocommerce/woocommerce
# Remove use of SQL_CALC_FOUND_ROWS in low stock query

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #38932

This PR changes the following.

- Removed use of `SQL_CALC_FOUND_ROWS` -- SQL_CALC_FOUND_ROWS can cause inconsistent performance depending on the # of rows or DB version. Some stores may even experience timeout. See https://www.percona.com/blog/to-sql_calc_found_rows-or-not-to-sql_calc_found_rows/ for more info.
- Added `products/count-low-in-stock` API endpoint and uses it from the frontend. The endpoint is faster than the existing `products/low-in-stock` endpoint as it doesn't query the data. 



### How to test the changes in this Pull Request:

1. Create a new JN site with this branch.
2. Install and activate WooCommerce Smooth Generator from https://github.com/woocommerce/wc-smooth-generator/releases
3. Go to `Tools -> Smooth Generator.`
4. Generate 100 products and orders.
5. Wait until products and orders are ready.
6. Go to `WooCommerce -> Home`. Complete the core profiler if necessary.
7. Click ellipsis button and click `Hide setup list.`
8. Confirm `Stock` tap appears and it's working as expected.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Remove use SQL_CALC_FOUND_ROWS from the query. SQL_CALC_FOUND_ROWS can cause inconsistent performance depending on the # of rows or DB version. 

#### Comment 

</details>
